src/core/settings/qgssettingsregistrycore.cpp

```diff
@@ -237,33 +237,40 @@ void QgsSettingsRegistryCore::migrateOldSettings()
   {
+    const QStringList existingConnections { QgsOwsConnection::connectionList( service.toLower() ) };
     QgsSettings settings;
     settings.beginGroup( QStringLiteral( "qgis/connections-%1" ).arg( service.toLower() ) );
     const QStringList connections = settings.childGroups();
     if ( connections.count() == 0 )
       continue;
-    for ( const QString &connection : connections )
+    for ( const QString &connection : std::as_const( connections ) )
     {
-      QgsOwsConnection::settingsUrl->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/url" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsVersion->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/version" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsIgnoreGetMapURI->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/ignoreGetMapURI" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsIgnoreGetFeatureInfoURI->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/ignoreGetFeatureInfoURI" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsSmoothPixmapTransform->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/smoothPixmapTransform" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsReportedLayerExtents->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/reportedLayerExtents" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsDpiMode->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/dpiMode" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsTilePixelRatio->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/tilePixelRatio" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsMaxNumFeatures->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/maxnumfeatures" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsPagesize->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/pagesize" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsPagingEnabled->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/pagingenabled" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsPreferCoordinatesForWfsT11->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/preferCoordinatesForWfsT11" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsIgnoreAxisOrientation->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/ignoreAxisOrientation" ), {service.toLower(), connection}, true );
-      QgsOwsConnection::settingsInvertAxisOrientation->copyValueFromKey( QStringLiteral( "qgis/connections-%1/%2/invertAxisOrientation" ), {service.toLower(), connection}, true );
+      // Do not migrate if new setting is already set
+      if ( existingConnections.contains( connection ) )
+      {
+        continue;
```

Comment:
I am not sure this is the right approach. The idea of the migration is forward and backward compatible. 
So, if you change something in an old version of QGIS it would also be updated in the latest version.

Reply:
Oh, I see, how does it work when the settings are modified in master? Are they written also in the old settings?

Reply:
yes, in backward compatibility 
https://github.com/qgis/QGIS/blob/9b9f431db10c47ae2233739f2292534af9cc5cdc/src/core/settings/qgssettingsregistrycore.cpp#L446

Reply:
ok, I'll have another look.